flink-connector-opensearch/src/main/java/org/apache/flink/connector/opensearch/sink/OpensearchSinkBuilder.java

```diff
@@ -0,0 +1,449 @@
+     *
+     * @param username of the Opensearch cluster user
+     * @return this builder
+     */
+    public OpensearchSinkBuilder<IN> setConnectionUsername(String username) {
+        checkNotNull(username);
+        this.username = username;
+        return self();
+    }
+
+    /**
+     * Sets the password used to authenticate the conection with the Opensearch cluster.
+     *
+     * @param password of the Opensearch cluster user
+     * @return this builder
+     */
+    public OpensearchSinkBuilder<IN> setConnectionPassword(String password) {
+        checkNotNull(password);
+        this.password = password;
+        return self();
+    }
+
+    /**
+     * Sets a prefix which used for every REST communication to the Opensearch cluster.
+     *
+     * @param prefix for the communication
+     * @return this builder
+     */
+    public OpensearchSinkBuilder<IN> setConnectionPathPrefix(String prefix) {
+        checkNotNull(prefix);
+        this.connectionPathPrefix = prefix;
+        return self();
+    }
+
+    /**
+     * Sets the timeout for requesting the connection of the Opensearch cluster from the connection
+     * manager.
+     *
+     * @param timeout for the connection request
+     * @return this builder
+     */
+    public OpensearchSinkBuilder<IN> setConnectionRequestTimeout(int timeout) {
+        checkState(timeout >= 0, "Connection request timeout must be larger than or equal to 0.");
+        this.connectionRequestTimeout = timeout;
+        return self();
+    }
+
+    /**
+     * Sets the timeout for establishing a connection of the Opensearch cluster.
+     *
+     * @param timeout for the connection
+     * @return this builder
+     */
+    public OpensearchSinkBuilder<IN> setConnectionTimeout(int timeout) {
+        checkState(timeout >= 0, "Connection timeout must be larger than or equal to 0.");
+        this.connectionTimeout = timeout;
+        return self();
+    }
+
+    /**
+     * Sets the timeout for waiting for data or, put differently, a maximum period inactivity
+     * between two consecutive data packets.
+     *
+     * @param timeout for the socket
+     * @return this builder
+     */
+    public OpensearchSinkBuilder<IN> setSocketTimeout(int timeout) {
+        checkState(timeout >= 0, "Socket timeout must be larger than or equal to 0.");
+        this.socketTimeout = timeout;
+        return self();
+    }
+
+    /**
+     * Allows to bypass the certificates chain validation and connect to insecure network endpoints
+     * (for example, servers which use self-signed certificates).
+     *
+     * @param allowInsecure allow or not to insecure network endpoints
+     * @return this builder
+     */
+    public OpensearchSinkBuilder<IN> setAllowInsecure(boolean allowInsecure) {
+        this.allowInsecure = allowInsecure;
+        return self();
+    }
+
+    protected BulkProcessorBuilderFactory getBulkProcessorBuilderFactory() {
```

Comment:
There's no reason for this to be protected since BulkProcessorBuilderFactory is an internal interface. 
Maybe that entire class shouldn't exist in it's current form, as it only existed to supported different elasticsearch versions (unless we intend to do the same in the near future for opensearch).

Reply:
> There's no reason for this to be protected since BulkProcessorBuilderFactory is an internal interface

:+1: 

> Maybe that entire class shouldn't exist in it's current form, as it only existed to supported different elasticsearch versions (unless we intend to do the same in the near future for opensearch).

Yes, it could be removed since the only place `BulkProcessor` is need is inside the `OpensearchWriter`